templates/ingress-gateways-deployment.yaml

```diff
@@ -110,6 +110,9 @@ spec:
           volumeMounts:
           - name: consul-bin
             mountPath: /consul-bin
+          {{- if ( default $root.Values.global.initCopyContainerResources .resources) }}
+          resources: {{ toYaml (default $root.Values.global.initCopyContainerResources.resources .resources) | nindent 12 }}
```

Comment:
```suggestion
          resources: 
            {{ toYaml (default $root.Values.global.initCopyContainerResources.resources .resources) | nindent 12 }}
```

Reply:
Thanks for the suggestion, oddly enough that is how I had it at first, and the formatting actually results in a yaml formatted like this : 
```
  resources:
  
    requests:
      memory: "100Mi"
       cpu: "100m"
....
```
Thoughts?

Reply:
That is surprising. I was looking at the rest of the diff, and it looked like it was set in a newline everywhere else. 🤯

Reply:
Actually, i think i see it maybe? 
`{{ toYaml (default .Values.global.initCopyContainerResources .resources) | nindent 12 }}` vs `{{ toYaml (default $root.Values.global.initCopyContainerResources.resources .resources) | nindent 12 }}`

One of them has a `.resources` at the end of the `initCopyContainerResources` and the other doesn't.

Reply:
The use of `default` isn't needed here. default is used in the other location to allow setting a default for the gateway container and overriding it subsequently. As written, if they set resources for the gateway container those settings will actually get used for the copy containers which isn't what we want, e.g.
```yaml
ingressGateways:
  resources:
    limits:
      memory: 1000m # this is now used for the copy, lifecycle AND the main envoy container.
```

In this case, because we don't allow users to override the copy container/lifecycle container's settings at all. We just need to reference the single global setting.

So this section should read
```yaml
          resources: {{ toYaml $root.Values.global.initCopyContainerResources.resources | nindent 12 }}
```

There's no need for the `if` either.